lib/features/past_search_cities/actions/add_past_search_city_action.dart

```diff
@@ -0,0 +1,16 @@
+import 'dart:async';
+
+import 'package:lesson1/app/app_state.dart';
+import 'package:lesson1/app/base_action.dart';
+
+class AddPastSearchCityAction extends BaseAction {
+  AddPastSearchCityAction(this.city);
+
+  final String city;
+
+  @override
+  Future<AppState?> reduce() async {
```

Comment:
```suggestion
  AppState? reduce() {
```

если нет асинхронных операций внутри экшена, то можно делать его синхронным как в этом случае)